core/src/main/java/jenkins/agents/WebSocketAgents.java

```diff
@@ -87,11 +88,11 @@ public HttpResponse doIndex(StaplerRequest req, StaplerResponse rsp) throws IOEx
         state.setRemoteEndpointDescription(req.getRemoteAddr());
         state.fireBeforeProperties();
         LOGGER.fine(() -> "connecting " + agent);
-        state.fireAfterProperties(ImmutableMap.of(
-            // TODO or just pass all request headers?
-            JnlpConnectionState.CLIENT_NAME_KEY, agent,
-            JnlpConnectionState.SECRET_KEY, secret
-        ));
+        // TODO or just pass all request headers?
+        Map<String, String> properties = new HashMap<>();
+        properties.put(JnlpConnectionState.CLIENT_NAME_KEY, agent);
+        properties.put(JnlpConnectionState.SECRET_KEY, secret);
+        state.fireAfterProperties(Collections.unmodifiableMap(properties));
```

Comment:
`properties` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.